application/src/main/java/org/togetherjava/tjbot/commands/TopHelpers.java

```diff
@@ -0,0 +1,114 @@
+package org.togetherjava.tjbot.commands;
+
+import com.github.freva.asciitable.HorizontalAlign;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.Record3;
+import org.jooq.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.util.PresentationUtils;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.jooq.impl.DSL.*;
+import static org.togetherjava.tjbot.db.generated.tables.MessageMetadata.MESSAGE_METADATA;
+import static org.togetherjava.tjbot.util.MessageTemplate.PLAINTEXT_MESSAGE_TEMPLATE;
+
+public final class TopHelpers implements Command {
+    private static final Logger logger = LoggerFactory.getLogger(TopHelpers.class);
+
+    private static final String COUNT_OPTION = "count";
+    private static final String NO_ENTRIES = "No entries";
+
+    private static final long MIN_COUNT = 10L;
+    private static final long MAX_COUNT = 30L;
+
+    private final Database database;
+
+    /**
+     * Initializes TopHelpers with a database.
+     *
+     * @param database the database to store the key-value pairs in
+     */
+    public TopHelpers(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public @NotNull String getCommandName() {
+        return "tophelpers";
+    }
+
+    @Override
+    public @NotNull String getDescription() {
+        return "Find top helpers for last 30 days";
+    }
+
+    @Override
+    public boolean isGuildOnly() {
+        return true;
+    }
+
+    @Override
+    public @NotNull CommandData addOptions(@NotNull CommandData commandData) {
+        return commandData.addOptions(new OptionData(OptionType.STRING, COUNT_OPTION,
+                "Count of top helpers to be retrieved (default is 10 and capped at 30)", false));
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandEvent event) {
+        var guildId = event.getGuild().getIdLong();
+        var count = getBoundedCount(event.getOption(COUNT_OPTION));
+        database.read(dsl -> {
+            var records = dsl.with("TOPHELPERS")
```

Comment:
See above, no idea what var is

Reply:
Yes you said it's your recommendation. Here records are well records it's an intermediary state. A type of `Result<Record3<Integer, Long, Long>>` isnt very helpful.

I mean you wouldn't be concerned about what's the type of `dsl.with()` or `select(..).as(...)` you are just making sense based on their names. Plus everything is in your IDE